### PR TITLE
zynqmp: updated conted based on new build makefile

### PR DESCRIPTION
**Pull request by ierkiaga**
A new build makefile has been merged in the build repository which is
based on open source repositories and buildroot, instead of Petalinux.

This updated content reflects these changes and redirects the build
instruction to the common "Get and build the solution" page.

Signed-off-by: Ibai Erkiaga <ibaie@xilinx.com>

### CODE REVIEW

File: building/devices/zynqmp.rst [hunk 1]
```diff
@@ -1,150 +1,57 @@
...
-	$ sudo e2image -rap /path/to/project/images/linux/rootfs.ext4 /dev/sdb2
++------------------------+--------------------------------------+
+| Board Name             | Build configuration directive        |
++========================+======================================+
+| ZCU102                 | ``PLATFORM=zynqmp-zcu102``           |
++------------------------+--------------------------------------+
 
-Now you can use the newly created SD card to boot your board.
+After completion of the buildling process, two new files will be generated within the zynqmp folder, BOOT.bin and <platform-name>.ub.
+The first one is the boot image compound by the FSBL, PMU Firmware, ARM Trusted Firmware, OP-TEE and U-Boot. The second one is a FIT image containing the Linux kernel, the device-tree blob and the initramfs root file system.
```
Comment (etienne-lms) [inline building/devices/zynqmp.rst:34]: s/compound by/composed of/?

File: building/devices/zynqmp.rst [hunk 2]
```diff
@@ -1,150 +1,57 @@
...
-	$ cp /path/to/project/images/linux/image.ub /dev/sdb1
-	$ sudo e2image -rap /path/to/project/images/linux/rootfs.ext4 /dev/sdb2
++------------------------+--------------------------------------+
+| Board Name             | Build configuration directive        |
++========================+======================================+
+| ZCU102                 | ``PLATFORM=zynqmp-zcu102``           |
++------------------------+--------------------------------------+
 
-Now you can use the newly created SD card to boot your board.
+After completion of the buildling process, two new files will be generated within the zynqmp folder, BOOT.bin and <platform-name>.ub.
```
Comment (etienne-lms) [inline building/devices/zynqmp.rst:33]: 80 char/line for the text. Please wrap.

Mark file names with "``":
```patch
- within the zynqmp folder, BOOT.bin and <platform-name>.ub.
+within the ``zynqmp/`` folder: ``BOOT.bin`` and ``<platform-name>.ub``. 
```

File: building/devices/zynqmp.rst [hunk 3]
```diff
@@ -1,150 +1,57 @@
...
+The first one is the boot image compound by the FSBL, PMU Firmware, ARM Trusted Firmware, OP-TEE and U-Boot. The second one is a FIT image containing the Linux kernel, the device-tree blob and the initramfs root file system.
 
 .. note::
+	If the firmware image is not provided to the build script the boot image will not be generated
```
Comment (etienne-lms) [inline building/devices/zynqmp.rst:37]: add a terminal dot.

**Review by etienne-lms — COMMENTED** [review]
In commit log:
* fix header line: s/conted/content/. Maybe: `zynqmp: update build instructions`.
* s/instruction/instructions/

File: building/devices/zynqmp.rst (hunk 3 above)
Comment (ierkiaga) [reply building/devices/zynqmp.rst:37]: Hi @etienne-lms, not sure what do you mean with "add a terminal dot" here :)

File: building/devices/zynqmp.rst (hunk 2 above)
Comment (ierkiaga) [reply building/devices/zynqmp.rst:33]: Is the wrapping really required? I mean, don't want to question if there is a guideline that requires it but I thought Sphinx would render the long lines appropriately. I have more lines >80 so I will wrap all of them if so

Comment (etienne-lms) [reply building/devices/zynqmp.rst:33]: > Is the wrapping really required?

Make it really easier for readers to follow the docs and adds consistency between articles in the docs.

File: building/devices/zynqmp.rst (hunk 3 above)
Comment (etienne-lms) [reply building/devices/zynqmp.rst:37]: End the sentence with a period `.`:
```
	If the firmware image is not provided to the build script the boot
        image will not be generated.

**Comment by etienne-lms** [comment]
Please fix commit log:
 * fix header line: s/`conted`/`content`/. Maybe rephrase: `zynqmp: update build instructions`?
 * replace `instruction` with `instructions`.

**Comment by ierkiaga** [comment]
Hi @etienne-lms , just a quick follow-up on this as github move it to stale status

File: building/devices/zynqmp.rst [hunk 4]
```diff
@@ -1,150 +1,74 @@
 .. _zynqmp:
 
 #########################
-ZynqMP zcu10x and Ultra96
+Zynq MPSoC
 #########################
```
Comment (etienne-lms) [inline building/devices/zynqmp.rst:5]: I think you need to update  `#####... ` to the section title size (I'm not sure but it's the case on other .rst files).

```
##########
Zynq MPSoC
##########
(...)

Supported boards
****************
(...)

File: building/devices/zynqmp.rst [hunk 5]
```diff
@@ -1,150 +1,74 @@
...
-bootable SD card. To run QEMU:
-
-.. code-block:: bash
-
-	$ make -f zynqmp.mk qemu
++--------------+--------------+----------------------+
+| Board Name   | Manufacturer | Hardware Description |
++==============+==============+======================+
+| ZCU102       | Xilinx       | `ZCU102 Website`_    |
++--------------+--------------+----------------------+
```
Comment (etienne-lms) [inline building/devices/zynqmp.rst:14]: This change removes build instructions for board zcu104, zcu106 and Ultra96v1.
@mimok, @ricardosalveti, are you ok with the proposed change in the documentation?

Comment (ricardosalveti) [reply building/devices/zynqmp.rst:14]: Yeah, please don't remove the build instructions for the other targets, are platform_config.h has the right checks based on the flavor used (e.g. ultra96 uses a different uart base address).

It is fine to have a more generic build instruction for the ZCU102 family, but also leave the part describing for other ZynqMP boards.

Comment (ricardosalveti) [reply building/devices/zynqmp.rst:14]: The instructions for the other targets should be similar, only PLATFORM will be different. Just don't remove the older block that says how to build for the other targets and we should be good.
